plugins/uefi-capsule/efi/generate_binary.py

```diff
@@ -95,6 +93,17 @@ def _run_objcopy(args):
             ".dynsym",
             "-j",
             ".rel*",
+        ]
+        if args.sbat_distro_id:
+            argv += [
+                "-j",
+                ".sbat",
+                "--set-section-alignment",
+                ".sbat=512",
+                "--change-section-vma",
+                ".sbat=0x2000",
+            ]
```

Comment:
The changes looks good to me and it leads to a binary with proper alignment and a VMA that doesn't cause issues for `edk2`. I would rather prefer this small change than pulling all the linker scripts, but wasn't sure if it was a safe enough option to make these assumptions about the default `gnu-efi` linker script.

As mentioned in the issue, if these values are not good for any distro then probably we should provide some meson options to set them if that was the case. I would like to know @vathpela @chrisccoulson option on this.

Reply:
This will probably create a working amd64 binary, but the aarch64 linker script places the .text section first. I've just cross-compiled shim for aarch64 and the .text section ends up at 0x0 in the ELF binary and 0x1000 in the final PE binary, so this would require a different solution for aarch64.

Reply:
Urgh, how complicated... so that header that edk2 places at VMA 0x0 is not added for aarch64 ?

Reply:
This is the reason why I think all this logic has to be handled in the system linker scripts instead of each project having to deal with it.

Reply:
I was confused about how the .text section at offset 0x0 in the ELF binary ended up at 0x1000 in the final PE binary, but it's because gnu-efi defines the PE header in assembly at the start of the ELF .text section to work around the lack of native support in binutils. So this isn't going to work on aarch64 at all right now :/

Reply:
I guess that's the nail in the coffin of this PR. Thanks for the explanation!